projects/angular/components/ui-grid/src/ui-grid.component.html

```diff
@@ -379,6 +383,15 @@
 </ng-template>
 
 <ng-template #noData>
+    <ng-container *ngIf="!loading">
+        <ng-container *ngIf="noContent; else defaultNoData">
+            <ng-container *ngTemplateOutlet="noContent.html">
```

Comment:
Should we add some context to the no data template for easy consumption?

```suggestion
            <ng-container *ngTemplateOutlet="noContent.html; context: { search: 'John', appliedFilters: 0 }">
```